libvast/vast/transform.hpp

```diff
@@ -9,9 +9,22 @@
 #pragma once
 
 #include "vast/transform_step.hpp"
+#include "vast/type.hpp"
+
+#include <queue>
 
 namespace vast {
 
+class offset_range {
+public:
+  void add(vast::id offset, table_slice::size_type rows);
+  bool contains(vast::id offset, table_slice::size_type rows);
+  void clear();
+
+private:
+  std::vector<std::pair<vast::id, table_slice::size_type>> ranges_;
+};
```

Comment:
Conceptually you probably want to use a `std::vector<id_range>` instead of this custom construct.